### PR TITLE
Feature pyoptsparse

### PR DESCRIPTION
**Pull request by planes**
This pull request implements pyoptsparse as an option as well as normal pyopt. Coveralls is modified to ignore the setup files for pyopt, pyoptsparse, and ipopt.

# Pull Request Checklist
 * Merge in the latest Develop branch changes to your branch
 * Remove .pyc files from your repository:
   * Linux/Mac
     * find . -name *.pyc -delete
   * Windows
     * del /S *.pyc
 * Run automatic regression and makes sure everything is passing
 * Did you make a new regression test that covers your new code?
 * Did you update your docstrings?
 * Did you update your headers to include your name and date?
 * Do a final compare with the Develop branch to be sure what you're changing

### CODE REVIEW

**Comment by planes** [comment]
I seriously have no idea how this decreased coverage 9.5%

**Comment by WallyMaier** [comment]
booo coverage

**Comment by planes** [comment]
Oh man, I just keep making this worse!

**Comment by planes** [comment]
BTW, you can run coveralls from your own computer without running the regression. Just install coverage:

 pip3 install coveralls

Then run coverage and check regressions: 

coverage run automatic_regression.py test

View the results by: 

coverage report -m

File: regression/.coveragerc [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    # Don't complain about missing debug-only code:
+    def __repr__
+    if self\.debug
+
+    # Don't complain if tests don't hit defensive assertion code:
+    raise AssertionError
+    raise NotImplementedError
+
+    # Don't complain if non-runnable code isn't run:
```
Comment (WallyMaier) [inline regression/.coveragerc]: I like this file.  However does this code skip code that may need to be regressed?

Comment (planes) [reply regression/.coveragerc]: I'm actually not certain about what this code does. These exclude lines are part of the starter code.

**Pull request merged**